node/tests/evm_burn_token.rs

```diff
@@ -0,0 +1,127 @@
+#![cfg(unix)]
+
+use assert_cmd::cargo::cargo_bin;
+use sp_core::U256;
+use std::process::Command;
+use tempfile::tempdir;
+
+use nix::{
+	sys::signal::{kill, Signal::SIGINT},
+	unistd::Pid,
+};
+
+use std::process::{self};
+pub mod common;
+
+use ethers::{
+	prelude::*,
+	providers::{Http, Provider},
+	utils,
+};
+
+type Client = SignerMiddleware<Provider<Http>, Wallet<k256::ecdsa::SigningKey>>;
+
+// Sends some native currency
+async fn send_transaction(
+	client: &Client,
+	address_from: &Address,
+	address_to: &Address,
+) -> Result<TransactionReceipt, Box<dyn std::error::Error>> {
+	println!(
+		"Beginning transfer of 10000 native currency {} to {}.",
+		address_from, address_to
+	);
+	let tx = Eip1559TransactionRequest::new()
+		.to(address_to.clone())
+		.value(U256::from(utils::parse_ether(10000)?))
+		.from(address_from.clone());
+	let tx = client.send_transaction(tx, None).await?.await?;
+
+	println!("Transaction Receipt: {}", serde_json::to_string(&tx)?);
+
+	Ok(tx.unwrap_or_default())
+}
+
+// Print the balance of a wallet
+async fn _print_balances(
+	provider: &Provider<Http>,
+	address_from: &Address,
+	address_to: &Address,
+) -> Result<(), Box<dyn std::error::Error>> {
+	let balance_from = provider.get_balance(address_from.clone(), None).await?;
+	let balance_to = provider.get_balance(address_to.clone(), None).await?;
+
+	println!("{} has {}", address_from, balance_from);
+	println!("{} has {}", address_to, balance_to);
+	Ok(())
+}
+
+#[tokio::test]
+async fn evm_burn_token_test() -> Result<(), Box<dyn std::error::Error>> {
+	let base_path = tempdir().expect("could not create a temp dir");
+
+	let mut cmd = Command::new(cargo_bin("golden-gate-node"))
+		.stdout(process::Stdio::piped())
+		.stderr(process::Stdio::piped())
+		.args(&["--dev"])
+		.arg("-d")
+		.arg(base_path.path())
+		.spawn()
+		.unwrap();
+
+	let stderr = cmd.stderr.take().unwrap();
+
+	let (ws_url, http_url, _) = common::find_ws_http_url_from_output(stderr);
+
+	let mut child = common::KillChildOnDrop(cmd);
+
+	// Let it produce some blocks.
+	let _ = common::wait_n_finalized_blocks(1, 30, &ws_url).await;
+
+	assert!(
+		child.try_wait().unwrap().is_none(),
+		"the process should still be running"
+	);
+
+	let provider: Provider<Http> = Provider::<Http>::try_from(http_url)?; // Change to correct network
+																	  // Do not include the private key in plain text in any produciton code. This is just for demonstration purposes
```

Comment:
Important comment but this is some very long whitespace... maybe run `rustfmt`?

Reply:
ok